src/ZoneServer/World/Actors/Characters/Character.cs

```diff
@@ -705,6 +711,12 @@ public void Heal(float hpAmount, float spAmount)
 			if (hpAmount == 0 && spAmount == 0)
 				return;
 
+			// 30% of healing reduced
```

Comment:
Putting this in the Heal method is potentially problematic. It's convenient that it works automatically on everything that heals, but at the same time you might call this method expecting to get exactly what you put in, and then be confronted with unexpected side effects. I believe we would need both, and at the _very_ least it should be documented.

Reply:
which side effects are you thinking on?

Reply:
The side effect I was referring to is that you're changing the amount of healing received. As a hypothetical example, let's say I wanted to create an NPC that heals the character for exactly 50% of their HP and SP. The way I would try to do that might be to get the max HP and SP, half them, and then call `Heal` with those values. This addition of yours would change the amount of HP the character is healed by, and the users who called the function might rightfully wonder why they didn't get the full amount. And since you didn't document this behavior in any way, i.e. didn't change the comments on the `Heal` method, the only way for them to figure out what happened is to dig into the source code, which is never ideal.